layouts/layout-1d-base.js

```diff
@@ -13,8 +13,6 @@ export default class Layout extends EventTarget {
     this._itemSize = {width: 100, height: 100};
     this._spacing = 0;
 
-    this._virtualScroll = false;
```

Comment:
This was a flag enabling the layout to be used with a non-native scroller.

When `_virtualScroll` was set to `false` (the default, and the current behavior), the layout was responsible for positioning elements within the "scrolling surface", assuming that the native scroller was moving the entire surface relative to the viewport.

When `_virtualScroll` was set to `true`, the layout took this to mean that the "scrolling surface" was virtual and instead positioned elements relative to the viewport itself (creating the illusion that the scrolling surface existed). In this mode, the assumption was that things like scroll physics and scroll thumb (normally provided by the native scroller) were being handled in JavaScript.

Reply:
Oh interesting, `virtualScroll = true` should then help reduce the surface required to perform the scroll, at the expense of having a custom scroll thumb which needs to be updated.
Is this an issue? I think css containment should already help on this, right?

Btw I searched usages of `_virtualScroll` in the original implementation but could not find any except for getter/setter https://github.com/graynorton/experimental-mono/search?q=_virtualScroll&unscoped_q=_virtualScroll

Reply:
Yeah, I seem to have removed that functionality at some earlier point but missed removing the getter and setter. I can't find the specific commit, but this earlier version of the layout seems to have the functionality still in place (it was only a few lines):

https://github.com/graynorton/experimental-mono/blob/d38147bd5e609edb6be336354806ede03b521d8d/packages/virtuality/layouts/layout-1d.js

Reply:
To answer your other question, I don't think we need this functionality now, and it would be trivial to re-add at some future point if necessary. It was mainly a vestige from predecessor implementations, targeting platforms where native scrolling within `overflow: auto` elements did not work well in various ways (no scroll physics, insufficient / buggy scroll events, etc.) and/or suffered from major performance issues when the size of the scrolling surface got really large. In these cases, it was necessary to bypass the native scroller and implement custom JS scrolling instead.